### PR TITLE
Release v4.0.0-rc1

### PR DESCRIPTION
**Pull request by ebfull**
This postpones `boost` and `native_rust` updates. (`native_rust` was renamed from `rust` in #4656)

### CODE REVIEW

File: doc/release-notes/release-notes-4.0.0-rc1.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
+Notable changes
```
Comment (str4d) [inline doc/release-notes/release-notes-4.0.0-rc1.md:1]: Missing: that this release will activate Canopy. Maybe others? Add it to both this file and the regular release-notes.md so it ends up in the final release.

File: doc/man/zcash-tx.1 [hunk 2]
```diff
@@ -1,9 +1,9 @@
-.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.47.13.
-.TH ZCASH-TX "1" "July 2020" "zcash-tx v3.1.0" "User Commands"
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.47.16.
+.TH ZCASH-TX "1" "August 2020" "zcash-tx v4.0.0-rc1" "User Commands"
 .SH NAME
-zcash-tx \- manual page for zcash-tx v3.1.0
+zcash-tx \- manual page for zcash-tx v4.0.0-rc1
```
Comment (defuse) [inline doc/man/zcash-tx.1:4]: It looks like either the first `-` shouldn't be escaped or the second `-` should be.

File: doc/man/zcashd.1 [hunk 3]
```diff
@@ -1,9 +1,9 @@
-.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.47.13.
-.TH ZCASHD "1" "July 2020" "zcashd v3.1.0" "User Commands"
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.47.16.
+.TH ZCASHD "1" "August 2020" "zcashd v4.0.0-rc1" "User Commands"
 .SH NAME
-zcashd \- manual page for zcashd v3.1.0
+zcashd \- manual page for zcashd v4.0.0-rc1
```
Comment (defuse) [inline doc/man/zcashd.1:4]: Same here.

**Comment by defuse** [comment]
This has an ACK from me after the two comments I left are addressed.

File: doc/man/zcash-tx.1 (hunk 2 above)
Comment (ebfull) [reply doc/man/zcash-tx.1:4]: This is just what the script outputs and so it's always been like this (for example, [see 3.1.0-rc1](https://github.com/zcash/zcash/commit/e255e8bd0a79326fc1827574a85448bd9e9f8b51#diff-3020b72dc7023f5e79b1ca6b5c03ad1dR4)), I think you're right though. I opened an issue: #4689

File: doc/man/zcashd.1 (hunk 3 above)
Comment (ebfull) [reply doc/man/zcashd.1:4]: Opened #4689

File: doc/release-notes.md [hunk 4]
```diff
@@ -4,6 +4,34 @@ release-notes at release time)
...
+already been activated on the Zcash testnet. Any node version 3.1.0 or higher,
+including this release, supports the Canopy activation on testnet.
+
+Disabling new value in the Sprout value pool
+--------------------------------------------
+
+After the mainnet activation of Canopy, it will not be possible to send funds to
+Sprout z-addresses from any _other_ kind of address, as described in [ZIP 211](https://zips.z.cash/zip-0211).
+It will still be possible to send funds _from_ a Sprout z-address and to send
+funds between Sprout addresses. Users of Sprout z-addresses are encouraged to
```
Comment (daira) [inline doc/release-notes.md:30]: [ZIP 212](https://zips.z.cash/zip-0211#specification) currently says:
> When this proposal is activated, nodes and wallets MUST disable any facilities to send to Sprout addresses, and this SHOULD be made clear in user interfaces and API documentation.

NCC raised this as an ambiguity in the ZIP:

image

If the intent is that fully shielded Sprout->Sprout and/or partially shielded Sprout -> Sprout+{transparent, Sapling} transactions are allowed, I can change the ZIP to say that. We need to check what is actually allowed by the wallet code.

File: doc/release-notes.md [hunk 5]
```diff
@@ -4,6 +4,34 @@ release-notes at release time)
 Notable changes
 ===============
 
+The mainnet activation of the Canopy network upgrade is supported by this
```
Comment (daira) [inline doc/release-notes.md:7]:
```suggestion
The mainnet activation of the Canopy network upgrade is supported by the 4.0.0
```

File: doc/release-notes/release-notes-4.0.0-rc1.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+===============
+
+The mainnet activation of the Canopy network upgrade is supported by this
```
Comment (daira) [inline doc/release-notes/release-notes-4.0.0-rc1.md:4]:
```suggestion
The mainnet activation of the Canopy network upgrade is supported by the 4.0.0
```

File: doc/release-notes/release-notes-4.0.0-rc1.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+In order to help the ecosystem prepare for the mainnet activiation, Canopy has
+already been activated on the Zcash testnet. Any node version 3.1.0 or higher,
+including this release, supports the Canopy activation on testnet.
+
+Disabling new value in the Sprout value pool
+--------------------------------------------
+
+After the mainnet activation of Canopy, it will not be possible to send funds to
+Sprout z-addresses from any _other_ kind of address, as described in [ZIP 211](https://zips.z.cash/zip-0211).
+It will still be possible to send funds _from_ a Sprout z-address and to send
```
Comment (daira) [inline doc/release-notes/release-notes-4.0.0-rc1.md:26]: If `release-nodes.md` is changed, don't forget to change this as well.

File: doc/release-notes/release-notes-4.0.0-rc1.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+      depends: clang 6.0.1
+      depends: native_cctools 921, ld64 409.12, libtapi 1000.10.8
+      build: use macOS 10.14 SDK
+      doc: explain why passing -mlinker-version is required
+
+noname45688@gmail.com (2):
+      Updating to Python 3
+      Update to Python 3
+
+Jack Grigg (2):
```
Comment (daira) [inline doc/release-notes/release-notes-4.0.0-rc1.md:256]: Not sure why this wasn't merged with the other "Jack Grigg" section above.

File: doc/release-notes/release-notes-4.0.0-rc1.md [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+      debian: Rename X11 to Expat-with-advertising-clause in copyright
+      Adjust GetActiveFundingStream* comments
+
+teor (1):
+      Fix a comment typo in pow.cpp
+
+therealyingtong (7):
```
Comment (daira) [inline doc/release-notes/release-notes-4.0.0-rc1.md:263]: This and "ying tong" should be added as aliases for "Ying Tong Lai".

File: doc/release-notes/release-notes-4.0.0-rc1.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+      Define PRF diversifiers in prf.h
+      assert(leadbyte == 0x02) after every if(leadbyte != 0x01)
+
+ying tong (4):
+      Update doc/release-notes/release-notes-3.1.0.md
+      Make sure garbage bytes are different
+      Rename PRV_DIVERSIFIER to PRF_TAG
+      Add link to ZIP212 in coinbase comment
+
+zancas (1):
```
Comment (daira) [inline doc/release-notes/release-notes-4.0.0-rc1.md:278]: Add as an alias for "Zancas Wilcox".

File: doc/release-notes/release-notes-4.0.0-rc1.md [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+      Implement system for postponing dependency updates.
+      Change release instructions to block the release when dependencies are not updated and not postponed.
+      Enforce pre-release dependency update check in make-release.py
+      Extend deadline for postponing dependency updates
+      Add new dependencies to updatecheck.py, add a flag we can use to have our CI test it.
+
+Ariel Gabizon (1):
+      explain expiry error
+
+bambam (2):
```
Comment (daira) [inline doc/release-notes/release-notes-4.0.0-rc1.md:233]: Add as an alias for "Benjamin Winston".

**Review by daira — APPROVED** [review]
ACK modulo comments

File: doc/release-notes.md (hunk 4 above)
Comment (daira) [reply doc/release-notes.md:30]: I will fix the ZIP, but the release notes look correct here.

File: doc/release-notes/release-notes-4.0.0-rc1.md (hunk 8 above)
Comment (ebfull) [reply doc/release-notes/release-notes-4.0.0-rc1.md]: Did it manually for now.

**Comment by ebfull** [comment]
@zkbot r+

**Pull request merged**